examples/shader/post_process_pass.rs

```diff
@@ -0,0 +1,383 @@
+
+        // We need to get the render app from the main app
+        let Ok(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        // The renderer has multiple stages.
+        // For more details of each stage see the docs for `RenderSet`
+        render_app.init_resource::<PostProcessPipeline>();
+
+        // Create our node with the render world
+        let node = PostProcessNode::new(&mut render_app.world);
+
+        // Get the render graph for the entire app
+        let mut graph = render_app.world.resource_mut::<RenderGraph>();
+
+        // Get the render graph for 3d entities
+        let core_3d_graph = graph.get_sub_graph_mut(core_3d::graph::NAME).unwrap();
+
+        // Register the post process node in the 3d render graph
+        core_3d_graph.add_node(PostProcessNode::NAME, node);
+
+        // A slot edge tells the render graph which input/output value should be passed to the node.
+        core_3d_graph.add_slot_edge(
+            core_3d_graph.input_node().id,
+            core_3d::graph::input::VIEW_ENTITY,
+            PostProcessNode::NAME,
+            PostProcessNode::IN_VIEW,
+        );
+
+        // We now need to add an edge between our node and the nodes from bevy
+        // to make sure our node is scheduled correctly.
+        //
+        // Here we want our effect to run after tonemapping and before the end of the main pass
+        core_3d_graph.add_node_edge(core_3d::graph::node::TONEMAPPING, PostProcessNode::NAME);
+        core_3d_graph.add_node_edge(
+            PostProcessNode::NAME,
+            core_3d::graph::node::END_MAIN_PASS_POST_PROCESSING,
+        );
+    }
+}
+
+/// The post process node used for the render graph
+struct PostProcessNode {
+    // The node needs a query to know how to render,
+    // but it's not a normal system so we need to define it manually.
+    query: QueryState<&'static ViewTarget, With<ExtractedView>>,
+}
+
+impl PostProcessNode {
+    pub const IN_VIEW: &str = "view";
+    pub const NAME: &str = "post_process";
+
+    fn new(world: &mut World) -> Self {
+        Self {
+            query: QueryState::new(world),
+        }
+    }
+}
+
+impl Node for PostProcessNode {
+    // This defines the input slot of the node and tells the render graph what
+    // we will need when running the node.
+    fn input(&self) -> Vec<SlotInfo> {
+        // In this case we tell the graph that our node will use the view entity.
+        // Currently, every node in bevy uses this pattern, so it's safe to just copy it.
+        vec![SlotInfo::new(PostProcessNode::IN_VIEW, SlotType::Entity)]
+    }
+
+    // This will run every frame before the run() method
+    // The important difference is that `self` is `mut` here
+    fn update(&mut self, world: &mut World) {
+        // Since this is not a system we need to update the query manually.
+        self.query.update_archetypes(world);
+    }
+
+    // Runs the node logic
```

Comment:
Mention how unlike queries that iterate over all relevant views at once, the node is invoked separately once per view. But word it better than how I just stated it :P